### PR TITLE
Ddf 1487 added audit logging capabilities.

### PR DESCRIPTION
**Pull request by ricklarsen**
[Review on Reviewable](https://reviewable.io/reviews/codice/ddf/217)

@stustison @vinamartin @clockard @djblue

### CODE REVIEW

**Comment by kcwire** [comment]
Please provide a title for the PR

File: catalog/security/catalog-security-auditplugin/src/main/resources/OSGI-INF/blueprint/blueprint.xml [hunk 1]
```diff
@@ -14,8 +14,8 @@
 -->
 <blueprint xmlns="http://www.osgi.org/xmlns/blueprint/v1.0.0">
 
-    <bean id="auditPlugin" class="ddf.catalog.security.audit.RetrieveResourceAuditPlugin"/>
+    <bean id="auditPlugin" class="ddf.catalog.security.audit.AuditPlugin"/>
 
-    <service ref="auditPlugin" auto-export="interfaces"/>
+    <service ref="auditPlugin" auto-export="interfaces" ranking="0"/>
```
Comment (clockard) [inline catalog/security/catalog-security-auditplugin/src/main/resources/OSGI-INF/blueprint/blueprint.xml]: Should this have a high ranking? I am just thinking that if a plugin like the IngestPlugin (rank=100) would run first and could reject the request. Then this plugin would never get run and we wouldn't log the fact that someone tried to ingest.

File: catalog/security/catalog-security-filter/src/main/java/ddf/catalog/security/filter/plugin/FilterPlugin.java [hunk 2]
```diff
@@ -93,14 +94,15 @@ public QueryResponse process(QueryResponse input)
...
-                logger.debug("Filtering metacard {}", metacard.getId());
-                SecurityLogger.logInfo("Filtering metacard " + metacard.getId());
+                filteredMetacards ++;
             } else {
-                SecurityLogger.logInfo("Allowing metacard " + metacard.getId());
                 newResults.add(result);
             }
         }
 
+        logger.info("Filtered " + filteredMetacards + "metacards, returned " + (newResults.size() - filteredMetacards));
```
Comment (vinamartin) [inline catalog/security/catalog-security-filter/src/main/java/ddf/catalog/security/filter/plugin/FilterPlugin.java]: Probably need to add a space before the word metacards in the string literal.

Comment (ricklarsen) [inline catalog/security/catalog-security-filter/src/main/java/ddf/catalog/security/filter/plugin/FilterPlugin.java]: good catch. fixed.

File: catalog/security/catalog-security-auditplugin/src/main/resources/OSGI-INF/blueprint/blueprint.xml (hunk 1 above)
Comment (ricklarsen) [inline catalog/security/catalog-security-auditplugin/src/main/resources/OSGI-INF/blueprint/blueprint.xml]: fixed.

**Comment by clockard** [comment]
:+1:

**Comment by vinamartin** [comment]
:+1:

**Comment by clockard** [comment]
Ran prbs script on this pr with no errors. LGTM

**Comment by harrison-tarr** [comment]
Review status: 0 of 5 files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---

<sup>**[catalog/security/catalog-security-filter/src/main/java/ddf/catalog/security/filter/plugin/FilterPlugin.java, line 103 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/217#-K-UkG-s3P3Kc4s0sZkB-r1-103)** ([raw file](https://github.com/codice/ddf/blob/fd1a7359d8756b4894fc7e7639051db07cfcd73c/catalog/security/catalog-security-filter/src/main/java/ddf/catalog/security/filter/plugin/FilterPlugin.java#L103)):</sup>
there's another one that needs to be fixed, looks like line 103


---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/217)

**Comment by harrison-tarr** [comment]
Review status: 0 of 5 files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---

<sup>**[catalog/security/catalog-security-filter/src/main/java/ddf/catalog/security/filter/plugin/FilterPlugin.java, line 103 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/217#-K-UkG-s3P3Kc4s0sZkB-r1-103)** ([raw file](https://github.com/codice/ddf/blob/fd1a7359d8756b4894fc7e7639051db07cfcd73c/catalog/security/catalog-security-filter/src/main/java/ddf/catalog/security/filter/plugin/FilterPlugin.java#L103)):</sup>
also this `logger.info` could be parameterized, couldn't it?


---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/217)

**Comment by ricklarsen** [comment]
<sup>**[catalog/security/catalog-security-filter/src/main/java/ddf/catalog/security/filter/plugin/FilterPlugin.java, line 103 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/217#-K-UkG-s3P3Kc4s0sZkB-r1-103)** ([raw file](https://github.com/codice/ddf/blob/fd1a7359d8756b4894fc7e7639051db07cfcd73c/catalog/security/catalog-security-filter/src/main/java/ddf/catalog/security/filter/plugin/FilterPlugin.java#L103)):</sup>
fixed.


---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/217)

**Pull request merged**